### ISSUE TITLE
iOS: MGLMultiPolyine does not display on map

### ISSUE THREAD

**Issue by nitrag** [mapbox/mapbox-gl-native#7876]
Adding `MGLPolyline` works as expected. Adding `MGLMultiPolyline` appears to work fine but it is not visible on the map. See example project [here](https://dl.dropboxusercontent.com/u/16045125/Mapbox%20BugTest_012617.zip) to reproduce.

**Platform:** iOS
**Mapbox SDK version:** 3.4.0-beta*

![image](https://cloud.githubusercontent.com/assets/197010/22346285/7e98f582-e3d1-11e6-8b33-b49a0d1fe4dc.png)

The red and blue are MultiLineString, the black line is a standard Linestring. Only the LineString appears on the map.

![image](https://cloud.githubusercontent.com/assets/197010/22346312/a2255464-e3d1-11e6-82bd-738627637929.png)

FYI, per the config, all lines should appear blue on the map.
```
func mapView(_ mapView: MGLMapView, strokeColorForShapeAnnotation annotation: MGLShape) -> UIColor {
    return UIColor.blue
}
func mapView(_ mapView: MGLMapView, lineWidthForPolylineAnnotation annotation: MGLPolyline) -> CGFloat {
    return 4
}
```

I am parsing the GeoJSON manually and using `mapView.addAnnotation()`. The `MGLMultiPolyline` **does** get parsed/added. I also tried `MGLMultiPolylineFeature`

**Comment by 1ec5** [comment]
Indeed, this is a [documented](https://www.mapbox.com/ios-sdk/api/3.4.1/Classes/MGLMapView.html#/c:objc%28cs%29MGLMapView%28im%29addAnnotation:) limitation of the annotation API. For now, your best bet is to use a shape source.

**Comment by 1ec5** [comment]
If you need to use the annotation API, you can “flatten” multipolylines and other unsupported shape types using [this code](https://github.com/mapbox/mapbox-gl-native/blob/042831604bab0983262146047d6be0e453b0d55d/platform/macos/app/MapDocument.m#L21-L51), which appears in the macOS SDK demo application. Note that you’ll end up needing to treat each polyline within the multipolyline as a separate annotation.

**Comment by nitrag** [comment]
I'm now using DDS to accomplish this. Feel free to close unless you want to leave open for feature tracking.
